src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -183,6 +206,230 @@ public void initAverRestCommunicator() throws Exception {
+								slowShutterOn();
+							} else if (value.equals(SWITCH_STATUS_OFF)) {
+								slowShutterOff();
+							}
+							break;
+						}
+						default: {
+							break;
+						}
+					}
+					break;
+				}
+				case IMAGE_PROCESS: {
+					// RGain
+					if (splitProperty[1].equals(Command.RGAIN.getName() + RGainControl.UP.getName())) {
+						rGainUp();
+						break;
+					} else if (splitProperty[1].equals(Command.RGAIN.getName() + RGainControl.DOWN.getName())) {
+						rGainDown();
+						break;
+					}
+
+					// BGain
+					if (splitProperty[1].equals(Command.BGAIN.getName() + BGainControl.UP.getName())) {
+						bGainUp();
+						break;
+					} else if (splitProperty[1].equals(Command.BGAIN.getName() + BGainControl.DOWN.getName())) {
+						bGainDown();
+						break;
+					}
+
+					Command imageProcessCommand = Command.getByName(splitProperty[1]);
+					switch (imageProcessCommand) {
+						case WB_MODE: {
+							if (value.equals(WBMode.AUTO.getName())) {
+								wbAuto();
+							} else if (value.equals(WBMode.INDOOR.getName())) {
+								wbIndoor();
+							} else if (value.equals(WBMode.OUTDOOR.getName())) {
+								wbOutdoor();
+							} else if (value.equals(WBMode.ONE_PUSH_WB.getName())) {
+								wbOnePush();
+							} else if (value.equals(WBMode.MANUAL.getName())) {
+								wbManual();
+							}
+							break;
+						}
+						case WB_ONE_PUSH_TRIGGER: {
+							wbOnePushTrigger();
+							break;
+						}
+						default: {
+							break;
+						}
+					}
+					break;
+				}
+				case PAN_TILT_DRIVE: {
+					if (splitProperty[1].equals(Command.PAN_TILT_HOME.getName())) {
+						panTiltHome();
+						break;
+					} else if (splitProperty[1].equals(Command.SLOW_PAN_TILT.getName())) {
+						if (value.equals(SWITCH_STATUS_ON)) {
+							slowPanTiltOn();
+						} else if (value.equals(SWITCH_STATUS_OFF)) {
+							slowPanTiltOff();
+						}
+						break;
+					}
+
+					PanTiltDrive pantTiltDrive = PanTiltDrive.getByName(splitProperty[1]);
+					switch (pantTiltDrive) {
+						case UP: {
+							panTiltUp();
+							break;
+						}
+						case DOWN: {
+							panTiltDown();
+							break;
+						}
+						case LEFT: {
+							panTiltLeft();
+							break;
+						}
+						case RIGHT: {
+							panTiltRight();
+							break;
+						}
+						case UP_LEFT: {
+							panTiltUpLeft();
+							break;
+						}
+						case UP_RIGHT: {
+							panTiltUpRight();
+							break;
+						}
+						case DOWN_LEFT: {
+							panTiltDownLeft();
+							break;
+						}
+						case DOWN_RIGHT: {
+							pantTiltDownRight();
+							break;
+						}
+					}
+					break;
+				}
+				case PRESET: {
+					if (splitProperty[1].equals(PresetControl.SET.getName())) {
+						setPreset(Integer.parseInt(value));
+					} else if (splitProperty[1].equals(PresetControl.RECALL.getName())) {
+						recallPreset(Integer.parseInt(value));
+					}
+					break;
+				}
+				default: {
+					break;
+				}
+			}
+		} finally {
```

Comment:
I think we should have the catch block and throw CommandFailureException with a clear error message